types/audit-details.d.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+declare global {
+  module LH.Audit.Details {
+    export interface Filmstrip {
+      type: 'filmstrip';
+      scale: number;
+      items: {
+        timing: number;
+        timestamp: number;
+        data: string;
+      }[];
+    }
+
+    export interface Screenshot {
+      type: 'screenshot';
+      timestamp: number;
+      data: string;
+    }
+
+    export interface Opportunity {
+      type: 'opportunity';
+      overallSavingsMs: number;
+      overallSavingsBytes?: number;
+      headings: TableColumnHeading[];
+      items: OpportunityItem[];
+    }
+
+    export interface CriticalRequestChain {
+      type: 'criticalrequestchain';
+      longestChain: {
+        duration: number;
+        length: number;
+        transferSize: number;
+      };
+      chains: Audit.SimpleCriticalRequestNode;
+    }
+
+    // TODO(bckenny)
+    // export interface MultiCheck {}
+    // export interface Table {}
+    // export interface SomeKindOfHiddenThing
```

Comment:
these intentionally left in here? 

these would be like a foreign language to me if I were ever tasked with following up maybe we could flesh out the comment a tad? :D